presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPushdownLimitIntoWindow.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.metadata.ResolvedFunction;
+import io.prestosql.spi.block.SortOrder;
+import io.prestosql.sql.planner.OrderingScheme;
+import io.prestosql.sql.planner.Symbol;
+import io.prestosql.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.prestosql.sql.planner.plan.LimitNode;
+import io.prestosql.sql.planner.plan.TopNRowNumberNode;
+import io.prestosql.sql.planner.plan.ValuesNode;
+import io.prestosql.sql.planner.plan.WindowNode;
+import io.prestosql.sql.tree.QualifiedName;
+import io.prestosql.sql.tree.SymbolReference;
+import io.prestosql.sql.tree.WindowFrame;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.sql.analyzer.TypeSignatureProvider.fromTypes;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.node;
+import static io.prestosql.sql.tree.FrameBound.Type.CURRENT_ROW;
+import static io.prestosql.sql.tree.FrameBound.Type.UNBOUNDED_PRECEDING;
+
+public class TestPushdownLimitIntoWindow
+        extends BaseRuleTest
+{
+    @Test
+    public void testLimitAboveWindow()
+    {
+        ResolvedFunction rowNumber = tester().getMetadata().resolveFunction(QualifiedName.of("row_number"), fromTypes());
+        tester().assertThat(new PushdownLimitIntoWindow(tester().getMetadata()))
+                .on(p -> {
+                    Symbol a = p.symbol("a", BIGINT);
+                    Symbol rowNumber1 = p.symbol("row_number_1");
+                    OrderingScheme orderingScheme = new OrderingScheme(
+                            ImmutableList.of(a),
+                            ImmutableMap.of(a, SortOrder.ASC_NULLS_FIRST));
+                    return p.limit(3, p.window(
+                            new WindowNode.Specification(ImmutableList.of(a), Optional.of(orderingScheme)),
+                            ImmutableMap.of(rowNumber1, newWindowNodeFunction(rowNumber, "a")),
+                            p.values(a)));
+                })
+                .matches(node(LimitNode.class, node(TopNRowNumberNode.class, node(ValuesNode.class))));
```

Comment:
Please define the expected `TopNRowNumberNode ` more precisely. Use `PlanMatchPattern.topNRowNumber` to set the expected partitioning, ordering and row count.
You can refer to `TestPruneTopNRowNumberColumns` for example usage.
Please apply to other tests cases too.